### PR TITLE
adding a new endpoint for canceling claims to bip mocks

### PR DESCRIPTION
**Pull request by tejans24**
## What was the problem?
fixes: #2096 to create a Mock endpoint to support `/cliams/{claim_id}/cancel`

## How does this fix it?[^1]
Adds a cancel endpoint to the mock API. 

If claim not found Response:

```
404
{
  "messages": [
    {
        "key": null,
        "severity": null,
        "status": null,
        "text": "No claim found for id: 1015",
	"timestamp": null
     }
  ]
}
```

```
200
{
  "messages": [
    {
        "key": null,
        "severity": null,
        "status": null,
        "text": "Successfully canceled the claim with id: 1015",
	"timestamp": null
     }
  ]
}
```

## How to test this PR

I changed `SecurityConfig.java` locally to disable security:

```
package gov.va.vro.mockbipclaims.config;

import gov.va.vro.mockshared.rest.JwtRequestFilter;
import lombok.AllArgsConstructor;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.http.SessionCreationPolicy;
import org.springframework.security.web.AuthenticationEntryPoint;
import org.springframework.security.web.SecurityFilterChain;
import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;

@Configuration
@AllArgsConstructor
public class SecurityConfig {
  private final JwtRequestFilter filter;

  @Qualifier("delegatedAuthenticationEntryPoint")
  private final AuthenticationEntryPoint authEntryPoint;

  /**
   * Spring security to validate JWT.
   *
   * @param http HttpSecurity
   * @return SecurityFilterChain
   * @throws Exception Authentication failure
   */
  @Bean
  public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
    http.authorizeRequests()
      .antMatchers("/**").permitAll()
      .and()
    .csrf().disable();
    return http.build();
  }
}
```

Used http client to access PUT `http://localhost:20306/claims/1020/cancel`.

### CODE REVIEW

**Comment by tejans24** [comment]
@greene858 what's the best way to get accurate response details of 200s and 400s from BIP?

**Comment by greene858** [comment]
I used the openapi.json and the openapi-generator tool to generate CloseClaimRequest.java and CloseClaimResponse.java

**Comment by tejans24** [comment]
> I used the openapi.json and the openapi-generator tool to generate CloseClaimRequest.java and CloseClaimResponse.java

Yea, the open api spec mostly uses the generic `ProviderResponse`. Which doesn't provide details on the actual data that is returned in those objects from BIP within the messages:

For example:
```
"messages": [
    {
        "key": null,
        "severity": null,
        "status": null,
        "text": "Successfully canceled the claim with id: 1015",
	"timestamp": null
     }
  ]
```

File: mocks/mock-bip-claims-api/src/main/java/gov/va/vro/mockbipclaims/controller/ClaimsController.java [hunk 1]
```diff
@@ -34,4 +37,23 @@ public ResponseEntity<ClaimDetailResponse> getClaimById(Long claimId) {
 
     return new ResponseEntity<>(response, HttpStatus.OK);
   }
+
+  @Override
+  public ResponseEntity<CloseClaimResponse> cancelClaimById(
+      Long claimId, CloseClaimRequest closeClaimRequest) {
```
Comment (tejans24) [inline mocks/mock-bip-claims-api/src/main/java/gov/va/vro/mockbipclaims/controller/ClaimsController.java:43]: @dfitchett do you have any context on how this `CloseClaimRequest` is used within the cancel claim PUT request? What are they expecting in it?

**Review by msnwatson — APPROVED** [review]

**Review by chengjie8 — APPROVED** [review]
LGTM!

**Pull request merged**